### ISSUE TITLE
Go to definition on identifier in macro call fails

### ISSUE THREAD

**Issue by Trass3r** [clangd/clangd#459]
In GTest jumping to the fixture class from a test declaration does not work:

```cpp
class FixtureName ... {
....
};
TEST_F(FixtureN^ame, testname) ...

#define TEST_F(test_fixture, test_name)\
  GTEST_TEST_(test_fixture, test_name, test_fixture, \
              ::testing::internal::GetTypeId<test_fixture>())

#define GTEST_TEST_(test_suite_name, test_name, parent_class, parent_id)      \
...
  class GTEST_TEST_CLASS_NAME_(test_suite_name, test_name) : public parent_class {                                                 \
```
It jumps to the start of the TEST_F line or to the enclosing namespace.

**System information**
Output of `clangd --version`: 10.0.1
Editor/LSP plugin: VSCode

**Comment by HighCommander4** [comment]
This works for me in the clangd codebase (which also uses GTest).

We'll likely need a complete example to diagnose further.

**Comment by HighCommander4** [comment]
Actually, this might have been fixed in #234, and that fix just didn't make LLVM 10.

(Not sure how exactly to tell when LLVM 10 branched.)

**Comment by HighCommander4** [comment]
Ok, I've confirmed that the fix for #234 is not present on the LLVM 10.x branch. And based on the definition of `TEST_F`, this is definitely the sort of thing that would have been fixed by #234 (macro argument appears in expansion multiple times).

So, I'm going to close this as a dupe of #234 (but feel free to reopen if you still see this bug with clangd 11).
